CRM/Contribute/Tokens.php

```diff
@@ -194,7 +194,43 @@ public function evaluateToken(TokenRow $row, $entity, $field, $prefetch = NULL)
       $row->tokens($entity, $field, $fieldValue);
     }
     else {
-      $row->dbToken($entity, $field, 'CRM_Contribute_BAO_Contribution', $field, $fieldValue);
+      // This section is copied & pasted & can be cleaned up later.
+      // most if probably already done in this functions
+      $tokenEntity = $entity;
+      $tokenField = $field;
+      $baoName = $this->getBAOName();
+      $baoField = $field;
+      if ($fieldValue === NULL || $fieldValue === '') {
+        return $row->tokens($tokenEntity, $tokenField, '');
+      }
+
+      $fields = $this->getFieldMetadata();
+      if (!empty($fields[$baoField]['pseudoconstant'])) {
+        $options = $baoName::buildOptions($baoField, 'get');
+        return $row->format('text/plain')
+          ->tokens($tokenEntity, $tokenField, $options[$fieldValue]);
+      }
+
+      switch ($fields[$baoField]['type']) {
+        case \CRM_Utils_Type::T_DATE + \CRM_Utils_Type::T_TIME:
+          return $row->format('text/plain')
+            ->tokens($tokenEntity, $tokenField, \CRM_Utils_Date::customFormat($fieldValue));
+
+        case \CRM_Utils_Type::T_MONEY:
+          // Is this something you should ever use? Seems like you need more context
+          // to know which currency to use.
+          return $row->format('text/plain')
```

Comment:
The if-else further up exists mostly to circumvent the code ever reaching this so once I am comfortable with the test cover I will remove